number.go

```diff
@@ -618,3 +620,376 @@ func (n *Number) Le(value interface{}) *Number {
 
 	return n
 }
+
+// IsInt succeeds if number is a signed integer of the specified bit width
+// as an optional argument.
+//
+// Bits argument defines maximum allowed bitness for the given number.
+// If bits is omitted, boundary checks is omitted too.
+//
+// value should have numeric type convertible to float64. Before comparison,
+// it is converted to float64.
+//
+// Example:
+//
+//	number := NewNumber(t, 123)
+//	number.IsInt(32)
+//	number.IsInt(64)
+//	number.IsInt()
+func (n *Number) IsInt(bits ...int) *Number {
+	opChain := n.chain.enter("IsInt()")
+	defer opChain.leave()
+
+	if opChain.failed() {
+		return n
+	}
+
+	if len(bits) > 1 {
+		opChain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple bits arguments"),
+			},
+		})
+		return n
+	}
+
+	if math.IsNaN(n.value) {
+		opChain.fail(AssertionFailure{
+			Type:   AssertType,
+			Actual: &AssertionValue{n.value},
+			Errors: []error{
+				errors.New("expected: number is integer"),
+			},
+		})
+		return n
+	}
+
+	inum, acc := big.NewFloat(n.value).Int(nil)
+	if acc != big.Exact {
+		opChain.fail(AssertionFailure{
+			Type:   AssertType,
+			Actual: &AssertionValue{n.value},
+			Errors: []error{
+				errors.New("expected: number is integer"),
+			},
+		})
+		return n
+	}
+
+	if len(bits) > 0 {
+		bitSize := bits[0]
+
+		imax := new(big.Int)
+		imax.Lsh(big.NewInt(1), uint(bitSize-1))
+		imax.Sub(imax, big.NewInt(1))
+		imin := new(big.Int)
+		imin.Neg(imax)
+		imin.Sub(imin, big.NewInt(1))
+		if inum.Cmp(imin) < 0 || inum.Cmp(imax) > 0 {
+			opChain.fail(AssertionFailure{
+				Type:   AssertInRange,
+				Actual: &AssertionValue{n.value},
+				Expected: &AssertionValue{AssertionRange{
+					fmt.Sprintf("-2^%d", bitSize-1),
+					fmt.Sprintf("2^%d", bitSize-1),
+				}},
```

Comment:
Also I suggest to improve reporting this ranges like this:

```go
Expected: &AssertionValue{AssertionRange{
    Min: intBoundary{imin, -1, bitSize-1},
    Max: intBoundary{imax, +1, bitSize-1},
}}

....

type intBoundary struct {
    val *big.Int
    sign int
    bits int
}

func (b intBoundary) String() string {
    if sign > 0 {
        return fmt.Sprintf("2^%d-1 (%s)", b.bits, val)
    } else {
        return fmt.Sprintf("-2^%d (%s)", b.bits, val)
    }
}
```

Explanations:

* we will give the user both logical form (2^N) and actual numeric value
* we will reuse this logic for all functions
* since intBoundary implements Stringer interface, it will be handled specially by httpexpect.DefaultFormatter: it will omit quotes when including it in error messages; i.e., instead of `"-2^7 (-128)"` we will see `-2^7 (-128)`, which looks prettier